main.py

```diff
@@ -0,0 +1,18 @@
+import pathlib
+
+import click
+
+
+@click.group()
+def cli():
+    pass
```

Comment:
Could you explain the vision for this file?

Presumably cli uses the package objects so that way the CLI is abstracted away from the core package in favor of something more akin to a factory.

Reply:
Specifically, I want to get rid of argparse as a next step to sustainability.

Click offers a lot of cool stuff when building a cli.

Reply:
But I didn't want to do all that in this PR